### PR TITLE
cmd/stack output: Introduce a Writer abstraction

### PR DESCRIPTION
**Pull request by abhinav**
There's a bit of branching and duplication
between the JSON and the default case.

    (with --json + without) x (with an argument + without)

That branching will only get worse with the upcoming `--shell` flag
(#2632).

This change defines an stackOutputWriter interface
that handles the actual printing,
with two implementations: one for JSON, one for user-readable output.

Note that all writers currently write directly to stdout.
In a future change, these may begin using `io.Writer`s.

In service to #2632

### CODE REVIEW

**Comment by abhinav** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11951** <a href="https://app.graphite.dev/github/pr/pulumi/pulumi/11951" target="_blank">Graphite</a> 
    * **PR #11952** <a href="https://app.graphite.dev/github/pr/pulumi/pulumi/11952" target="_blank">Graphite</a> 
      * **PR #11953** <a href="https://app.graphite.dev/github/pr/pulumi/pulumi/11953" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/pulumi/pulumi/11953?utm_source=stack-comment).

File: pkg/cmd/pulumi/stack_output.go [hunk 1]
```diff
@@ -97,22 +104,14 @@ func (cmd *stackOutputCmd) Run(ctx context.Context, args []string) error {
...
 		} else {
 			return fmt.Errorf("current stack does not have output property '%v'", name)
 		}
-	} else if cmd.jsonOut {
-		if err := printJSON(outputs); err != nil {
-			return err
-		}
-	} else {
-		printStackOutputs(outputs)
+	} else if err := outw.WriteMany(outputs); err != nil {
```
Comment (Frassle) [inline pkg/cmd/pulumi/stack_output.go:113]: I'd split this just to make it clear this is a plain "else" of args =0
```suggestion
	} else { 
	  if err := outw.WriteMany(outputs); err != nil {
```

**Review by Frassle — APPROVED** [review]

File: pkg/cmd/pulumi/stack_output.go (hunk 1 above)
Comment (abhinav) [reply pkg/cmd/pulumi/stack_output.go:113]: Ack. I flip-flopped on this when writing it, and your comment confirms my suspicion that the other way was more obvious.

**Pull request merged**